dart/lib/src/transport/transport.dart

```diff
@@ -0,0 +1,131 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:meta/meta.dart';
+import 'package:sentry/src/utils.dart';
+
+import '../protocol.dart';
+import '../sentry_options.dart';
+import 'body_encoder_browser.dart' if (dart.library.io) 'body_encoder.dart';
+import 'header_builder_browser.dart' if (dart.library.io) 'header_builder.dart';
+
+typedef BodyEncoder = List<int> Function(
+  Map<String, dynamic> data,
+  Map<String, String> headers, {
+  bool compressPayload,
+});
+
+/// A transport is in charge of sending the event to the Sentry server.
+class Transport {
+  final SentryOptions _options;
+
+  @visibleForTesting
+  final Dsn dsn;
+
+  /// Use for browser stacktrace
+  final String origin;
+
+  /// Used by sentry to differentiate browser from io environment
+  final String platform;
+
+  final Sdk sdk;
```

Comment:
if they are already in the options, they dont need to be public global fields

Reply:
for now I just let the `dsn` which is used for a lot of tests ( marked as @visibleForTestings )